precovery/precovery_db.py

```diff
@@ -67,6 +68,43 @@ class FrameCandidate:
     dataset_id: str
 
 
+def sort_candidates(
+    candidates: List[Union[PrecoveryCandidate, FrameCandidate]]
+) -> List[Union[PrecoveryCandidate, FrameCandidate]]:
+    """
+    Sort candidates by ascending MJD. For precovery candidates, use the MJD of the observation.
+    For frame candidates, use the MJD at the midpoint of the exposure.
+
+    Parameters
+    ----------
+    candidates : List[Union[PrecoveryCandidate, FrameCandidate]]
+        List of candidates to sort.
+
+    Returns
+    -------
+    List[Union[PrecoveryCandidate, FrameCandidate]]
+        Sorted list of candidates.
+    """
+    mjds = []
+    observation_ids = []
+    for candidate_i in candidates:
+        if isinstance(candidate_i, PrecoveryCandidate):
+            mjds.append(candidate_i.mjd)
+            observation_ids.append(candidate_i.observation_id)
+        elif isinstance(candidate_i, FrameCandidate):
+            mjds.append(candidate_i.exposure_mjd_mid)
+            # Add empty observation ID since frame candidates don't have one
+            # Frames, so this is fine
+            observation_ids.append("")
+        else:
+            raise TypeError("Candidates must be PrecoveryCandidate or FrameCandidate")
+
+    cand_array = np.array(candidates)
+    df = pd.DataFrame({"mjd": mjds, "observation_id": observation_ids})
+    df.sort_values(by=["mjd", "observation_id"], inplace=True)
+    return cand_array[df.index.values].tolist()
```

Comment:
Of course, what you have is fine, and looks correct. But I'd just like to share the stdlib, non-pandas way of doing this, which is in one line:

```suggestion
    return sorted(candidates, lambda c: (c.mjd, c.observation_id) if instance(c, PrecoveryCandidate) else (c.exposure_mjd_mid, ""))
```

Reply:
If the one-liner is too crazy, you could do

```suggestion
    def sort_key(candidate):
        if isinstance(candidate, PrecoveryCandidate):
            return (candidate.mjd, candidate.observation_id)
        elif isinstance(candidate, FrameCandidate):
            return (candidate.exposure_mjd_mid, "")
        else:
            raise TypeError(f"Candidates must be PrecoveryCandidate or FrameCandidate, but got {type(candidate)}")

    return sorted(candidates, key=sort_key)
```

Reply:
Your solution is faster than pandas (silly in-function dataframe creation \*shakes fists\*)!

See: 
```python
%%timeit 
sort_candidates(cands)
```
`700 µs ± 3.68 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)`

```python
%%timeit 
sort_candidates_no_pandas(cands)
```
`770 ns ± 4.21 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)`

Reply:
Wow, a factor of 1000 faster too! This is truly a great day for the anti-pandas resistance